Support elementwise add / mul for [B, *] nested, [B, 1] dense (CUDA only)

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95620

Small hack to reuse the 3D custom kernel from #88289 for [B, *] nested, [B, 1] dense elementwise add / mul. Simply treat the inputs as [B, *, 1], [B, 1, 1]. This is added to satisfy an internal ask.

Future work: full general broadcasting support between mixed nested / dense.

cc @cpuhrsch @bhosmer @drisspg @mikaylagawarecki

jbschlosser committed:
Support elementwise add / mul for [B, *] nested, [B, 1] dense (CUDA only)

[ghstack-poisoned]

drisspg reviewed (commented):
How does autograd work with the broadcasting support?  I dont' remember if we added support for the esuhm(maybe we remove this name from the code, idk) case. But if so we could add an autograd test that covers

mikaylagawarecki commented:
there's no autograd support for this case, and yea the name should probably be removed from the code, good catch

jbschlosser committed:
Update on "Support elementwise add / mul for [B, *] nested, [B, 1] dense (CUDA only)"


Small hack to reuse the ESUHM kernel from #88289 for [B, *] nested, [B, 1] dense elementwise add / mul. Simply treat the inputs as [B, *, 1], [B, 1, 1]. This is added to satisfy an ask from the Ads team.

Future work: full general broadcasting support between mixed nested / dense.

cc cpuhrsch bhosmer drisspg mikaylagawarecki

[ghstack-poisoned]

drisspg reviewed (approved):
🎸

jbschlosser commented:
@pytorchbot merge
